.github/workflows/sgx_hw.yml

```diff
@@ -18,6 +18,79 @@ defaults:
 jobs:
   sgx-hw:
     runs-on: self-hosted
+    env:
+      VM_BASE_NAME: gha-sgx-hw-vm
+      FAASM_VERSION: 0.9.3
     steps:
       - name: "Check out the experiment-base code"
         uses: actions/checkout@v3
+        with:
+          repository: faasm/experiment-base
+      - name: "Create a unique VM name"
+        run: echo "VM_NAME=${VM_BASE_NAME}-${{ github.job }}-${{ github.run_id }}-${{ github.run_attempt }}" >> $GITHUB_ENV
+      - name: "Create and provision an SGX-enabled IceLake VM"
+        run: |
+          ./bin/inv_wrapper.sh vm.create --sgx --region eastus2 --name ${{ env.VM_NAME }}
+          ./bin/inv_wrapper.sh vm.inventory --prefix ${{ env.VM_NAME }}
+          ./bin/inv_wrapper.sh vm.setup
+      - name: "Fetch the branch code if necessary"
+        run: |
+          ./bin/inv_wrapper.sh vm.run-command \
+            --name ${{ env.VM_NAME }} \
+            --path "code/faasm" \
+            --cmd "git config remote.origin.fetch '+refs/heads/*:refs/remotes/origin/*'"
+          ./bin/inv_wrapper.sh vm.run-command \
+            --name ${{ env.VM_NAME }} \
+            --path "code/faasm" \
+            --cmd "git fetch origin ${{ github.ref }}:ci-branch && git checkout -f ci-branch"
+          ./bin/inv_wrapper.sh vm.run-command \
+            --name ${{ env.VM_NAME }} \
+            --path "code/faasm" \
+            --cmd "git submodule update --init -f"
+          ./bin/inv_wrapper.sh vm.run-command \
+            --name ${{ env.VM_NAME }} \
+            --path "code/faasm/clients/python" \
+            --cmd "git submodule update --init -f third-party/cpp"
+      - name: "Start the docker compose cluster"
+        run: |
+          ./bin/inv_wrapper.sh vm.run-command \
+            --name ${{ env.VM_NAME }} \
+            --path "code/faasm" \
+            --cmd "./bin/refresh_local.sh"
+          ./bin/inv_wrapper.sh vm.run-command \
+            --name ${{ env.VM_NAME }} \
+            --path "code/faasm" \
+            --cmd "FAASM_BUILD_MOUNT=/build/faasm FAASM_LOCAL_MOUNT=/usr/local/faasm FAASM_CLI_IMAGE=faasm/cli-sgx:${{ env.FAASM_VERSION }} docker compose up --no-recreate -d faasm-cli cpp python"
+      - name: "Cross-compile CPP functions"
+        run: |
+          ./bin/inv_wrapper.sh vm.run-command \
+            --name ${{ env.VM_NAME }} \
+            --path "code/faasm" \
+            --cmd "sudo rm -rf ./clients/cpp/venv && docker compose exec cpp ./bin/inv_wrapper.sh func.local libfake"
```

Comment:
We remove the `./venv`'s to force their re-creation with the first call to `./bin/inv_wrapper.sh`, and to avoid possible inconsitencies between the new commits and the docker images.